python/tests/test_common_estimator.py

```diff
@@ -425,6 +425,15 @@ def assert_model(model: SparkRapidsMLDummyModel) -> None:
         model_loaded = SparkRapidsMLDummyModel.load(model_path)
         assert_model(model_loaded)
 
+        dummy2 = dummy.copy()
+        assert dummy2.cuml_params["a"] == 100
+        with pytest.raises(
+            Exception,
+            match="assert {'a': 9876, 'k': 4, 'x': 40.0} == {'a': 100, 'k': 4, 'x': 40.0}",
+        ):
+            dummy2.fit(df, {dummy2.alpha: 9876})
+        assert dummy2.cuml_params["a"] == 9876
```

Comment:
Not clear this is desired behavior, that original is updated with overriding param values.

Reply:
I agree that this doesn't look right.  I'd expect something like the following:
```
assert dummy2.cuml_params["a"] == 100
model2 = dummy2.fit(df, {dummy2.alpha: 9876})
assert dummy2.cuml_params["a"] == 100
assert model2.cuml_params["a"] == 9876
```

Basically, the extra param map should produce a temporary copy of the estimator which is then fit on the dataframe to produce `model2`.  This is how the Spark implementation behaves.

Reply:
Also note that in this line, `dummy2.alpha` doesn't set the value, it's just a Param (descriptor) paired with a value, so it could easily also be `dummy.alpha: 9876`.  Either way, it wouldn't update `dummy` or `dummy2`, but the hidden/temporary copy of `dummy2`.
```
model2 = dummy2.fit(df, {dummy2.alpha: 9876})
```

Reply:
Yes, you're right. I changed the spark behaviors. I will fix that.